src/radical/pilot/agent/radical-pilot-agent-multicore.py

```diff
@@ -1170,6 +1179,190 @@ def corner2offset(self, block, corner):
+
+
+            #if 'rm_ip' not in self._cfg['lrms_info']:
+            #    raise RuntimeError('rm_ip not in lm_info for %s' \
+            #            % (self.name))
+
+            self._log.info('Checking rm_ip %s' % self._cfg['lrms_info']['lm_info']['rm_ip'])
+            self._rm_ip = self._cfg['lrms_info']['lm_info']['rm_ip']
+            self._service_url = self._cfg['lrms_info']['lm_info']['service_url']
+            self._rm_url = self._cfg['lrms_info']['lm_info']['rm_url']
+            self._client_node = self._cfg['lrms_info']['lm_info']['nodename']
+
+            sample_time = rpu.timestamp()
+            yarn_status = ul.urlopen('http://{0}:8088/ws/v1/cluster/scheduler'.format(self._rm_ip))
+
+            yarn_schedul_json = json.loads(yarn_status.read())
+
+            max_num_app = yarn_schedul_json['scheduler']['schedulerInfo']['queues']['queue'][0]['maxApplications']
+            num_app = yarn_schedul_json['scheduler']['schedulerInfo']['queues']['queue'][0]['numApplications']
+
+            #-----------------------------------------------------------------------
+            # Find out the cluster's resources
+            cluster_metrics = ul.urlopen('http://{0}:8088/ws/v1/cluster/metrics'.format(self._rm_ip))
+
+            metrics = json.loads(cluster_metrics.read())
+            self._mnum_of_cores = metrics['clusterMetrics']['totalVirtualCores']
+            self._mmem_size = metrics['clusterMetrics']['totalMB']
+            self._num_of_cores = metrics['clusterMetrics']['allocatedVirtualCores']
+            self._mem_size = metrics['clusterMetrics']['allocatedMB']
+
+            self.avail_app = {'apps':max_num_app - num_app,'timestamp':sample_time}
+            self.avail_cores = self._mnum_of_cores - self._num_of_cores
+            self.avail_mem = self._mmem_size - self._mem_size
+
+
+            return True
+        except:
+            return False
+
+
+    # --------------------------------------------------------------------------
+    #
+    def slot_status(self):
+        """
+        Finds how many spots are left free in the YARN scheduler queue and also
+        updates if it is needed..
+        """
+        #-------------------------------------------------------------------------
+        # As it seems this part of the Scheduler is not according to the assumptions
+        # made about slot status. Keeping the code commented just in case it is
+        # needed later either as whole or art of it.
+        sample = rpu.timestamp()
+        yarn_status = ul.urlopen('http://{0}:8088/ws/v1/cluster/scheduler'.format(self._rm_ip))
+        yarn_schedul_json = json.loads(yarn_status.read())
+
+        max_num_app = yarn_schedul_json['scheduler']['schedulerInfo']['queues']['queue'][0]['maxApplications']
+        num_app = yarn_schedul_json['scheduler']['schedulerInfo']['queues']['queue'][0]['numApplications']
+        if (self.avail_app['timestamp'] - sample)>60 and \
```

Comment:
I wanted to introduce a delay between updates without using sleep.

If there is another idea, I am all ears!